src/main/java/org/cbio/gdcpipeline/reader/ClinicalReader.java

```diff
@@ -121,20 +126,26 @@ private void addData(TcgaBcr tcgaBcr) {
     }
 
     private List<File> getClinicalFileList() throws ItemStreamException {
-        if (uuidToFilesMap.isEmpty()) {
-            if (LOG.isErrorEnabled()) {
-                LOG.error(" UUID to Files Map is Empty.");
+        List<File> clincalFiles = new ArrayList<>();
+        List<String> filenames = new ArrayList<>();
+        if(!gdcFileMetadatas.isEmpty()){
+            for (GdcFileMetadata data : gdcFileMetadatas) {
+                if (data.getType().equals(CommonDataUtil.GDC_TYPE.CLINICAL.toString())) {
+                    filenames.add(data.getFile_name());
+                }
             }
-            throw new ItemStreamException("Empty File Map");
         }
-        File source = new File(sourceDir);
-        List<File> fileNames = new ArrayList<>();
-        for (File file : source.listFiles()) {
-            if (file.getName().endsWith(".xml") && file.getName().contains("clinical")) {
-                fileNames.add(file);
+        for (String f : filenames) {
```

Comment:
Can you remove this for loop and just do the file exists check before you add the file to the filenames list? No need to iterate through the files twice.